packages/clay-form/src/Input.tsx

```diff
@@ -155,6 +156,97 @@ const GroupInsetItem = React.forwardRef<
 
 GroupInsetItem.displayName = 'ClayInputGroupInsetItem';
 
+interface IInlineTextProps
+	extends React.HTMLAttributes<
+		HTMLDivElement | HTMLSpanElement | HTMLLabelElement
+	> {
+	/**
+	 * The id of the component
+	 */
+	id?: string;
+
+	/**
+	 * Text that hints at the expected data type to be entered.
+	 */
+	placeholder?: string;
+
+	/**
+	 * A flag that sets the component as noneditable.
+	 */
+	readOnly?: boolean;
+
+	/**
+	 * Path to the spritemap that Icon should use when referencing symbols.
+	 */
+	spritemap?: string;
+}
+
+const InlineText = React.forwardRef<
+	HTMLDivElement | HTMLSpanElement | HTMLLabelElement,
+	IInlineTextProps
+>(
+	({
+		children,
+		className,
+		id,
+		placeholder,
+		readOnly,
+		spritemap,
+		...otherProps
+	}: IInlineTextProps) => {
```

Comment:
We need to pass the `ref` forward to the span element of the component and in this case we just need to set the ref type to `HTMLSpanElement` or just pass it to the root element which might be better for the developer.